nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateRecord.java

```diff
@@ -0,0 +1,649 @@
+                                    if (nullValue) {
+                                        writeFieldValue = null;
+                                    } else {
+                                        final String propertyValue = context.getProperty(writeFieldName).getValue();
+                                        writeFieldValue = getFakeData(propertyValue, faker);
+                                    }
+                                }
+
+                                recordEntries.put(writeFieldName, writeFieldValue);
+                            }
+                            record = new MapRecord(recordSchema, recordEntries);
+                            writer.write(record);
+                        }
+
+                        final WriteResult writeResult = writer.finishRecordSet();
+                        attributes.put("record.count", String.valueOf(writeResult.getRecordCount()));
+                        attributes.put(CoreAttributes.MIME_TYPE.key(), writer.getMimeType());
+                        attributes.putAll(writeResult.getAttributes());
+                        recordCount.set(writeResult.getRecordCount());
+                    }
+                } catch (final SchemaNotFoundException e) {
+                    throw new ProcessException(e.getLocalizedMessage(), e);
+                }
+            });
+        } catch (final Exception e) {
+            getLogger().error("Failed to process {}; will route to failure", flowFile, e);
+            if (e instanceof ProcessException) {
+                throw e;
+            } else {
+                throw new ProcessException(e);
+            }
+        }
+
+        flowFile = session.putAllAttributes(flowFile, attributes);
+        session.transfer(flowFile, REL_SUCCESS);
+
+        final int count = recordCount.get();
+        session.adjustCounter("Records Processed", count, false);
+
+        getLogger().info("Successfully generated {} records for {}", count, flowFile);
+    }
+
+    protected Map<String, String> getFields(ProcessContext context) {
+        return context.getProperties().entrySet().stream()
+                // filter non-null dynamic properties
+                .filter(e -> e.getKey().isDynamic() && e.getValue() != null)
+                // convert to Map of user-defined field names and types
+                .collect(Collectors.toMap(
+                        e -> e.getKey().getName(),
+                        e -> context.getProperty(e.getKey()).getValue()
+                ));
+    }
+
+    private Object getFakeData(String type, Faker faker) {
+
+        // Catch these two cases ahead of calling the "discovered" Faker method below in order to return a double instead of a String
+        if (FT_LATITUDE_ALLOWABLE_VALUE_NAME.equals(type)) {
+            return Double.valueOf(faker.address().latitude());
+        }
+        if (FT_LONGITUDE_ALLOWABLE_VALUE_NAME.equals(type)) {
+            return Double.valueOf(faker.address().longitude());
```

Comment:
This call fails on the Windows build. It looks like the Faker `Address.longitude()` method uses `String.format()` without a Locale, which would cause it default to the platform default. Unfortunately this results in a NumberFormatException when parsing with `Double.valueOf()`. Instead of trying to parsing the number, it seems better to just treat `latitude` and `longitude` as string types.